### PR TITLE
Remove duplicated code

### PR DESCRIPTION
**Pull request by twogee**

### CODE REVIEW

File: src/main/org/apache/tools/ant/listener/Log4jListener.java [hunk 1]
```diff
@@ -142,21 +142,17 @@ public void messageLogged(final BuildEvent event) {
...
             case Project.MSG_INFO:
                 log.info(event.getMessage());
                 break;
             case Project.MSG_VERBOSE:
-                log.debug(event.getMessage());
-                break;
             case Project.MSG_DEBUG:
                 log.debug(event.getMessage());
                 break;
+            case Project.MSG_ERR:
```
Comment (bodewig) [inline src/main/org/apache/tools/ant/listener/Log4jListener.java:155]: Personally I prefer the explicit case over a fallthrough to the default case.

File: src/main/org/apache/tools/ant/taskdefs/MacroInstance.java [hunk 2]
```diff
@@ -221,8 +221,6 @@ private String macroSubs(String s, Map<String, String> macroMapping) {
             }
         }
         switch (state) {
-            case STATE_NORMAL:
-                break;
```
Comment (bodewig) [inline src/main/org/apache/tools/ant/taskdefs/MacroInstance.java:225]: To me the explicit case makes it easier to see what happens than having to look up the default case.

File: src/main/org/apache/tools/ant/taskdefs/optional/jlink/ClassNameReader.java [hunk 3]
```diff
@@ -56,43 +56,40 @@
...
-                break;
-
-            case FIELDREF :
-            case METHODREF :
-            case INTERFACEMETHODREF :
-            case NAMEANDTYPE :
-                values[i] = data.readInt();
-                break;
-            default:
+                case UTF8 :
```
Comment (bodewig) [inline src/main/org/apache/tools/ant/taskdefs/optional/jlink/ClassNameReader.java:59]: why did the indentation change?

File: src/main/org/apache/tools/ant/taskdefs/optional/net/FTP.java [hunk 4]
```diff
@@ -2671,27 +2664,24 @@ public void execute() throws BuildException {
...
-                return CHMOD;
-            case "mkdir":
-                return MK_DIR;
-            case "rmdir":
-                return RM_DIR;
-            case "site":
-                return SITE_CMD;
-            default:
-                return SEND_FILES;
+                case "recv":
```
Comment (bodewig) [inline src/main/org/apache/tools/ant/taskdefs/optional/net/FTP.java:2667]: indentation?

File: src/main/org/apache/tools/bzip2/BlockSort.java [hunk 5]
```diff
@@ -894,8 +893,8 @@ private void mainQSort3(final CBZip2OutputStream.Data dataShadow,
                     stack_ll[sp] = m;
                     stack_hh[sp] = hi;
                     stack_dd[sp] = d;
-                    sp++;
                 }
+                sp++;
```
Comment (bodewig) [inline src/main/org/apache/tools/bzip2/BlockSort.java:897]: technically this is redundant code but as `sp` gets incremented several time inside of the else block I'd prefer to leave the code as it. It makes shape of the code blocks that assign to the stack arrays the same for all four occasions (the if and three times inside of the else block).

File: src/main/org/apache/tools/ant/listener/Log4jListener.java (hunk 1 above)
Comment (twogee) [reply src/main/org/apache/tools/ant/listener/Log4jListener.java:155]: It is still explicit...

File: src/main/org/apache/tools/ant/taskdefs/MacroInstance.java (hunk 2 above)
Comment (twogee) [reply src/main/org/apache/tools/ant/taskdefs/MacroInstance.java:225]: It is explicit now, just as in above case, only merged with default

File: src/main/org/apache/tools/ant/listener/Log4jListener.java (hunk 1 above)
Comment (bodewig) [reply src/main/org/apache/tools/ant/listener/Log4jListener.java:155]: not explicit enough for my taste :-)

File: src/main/org/apache/tools/ant/taskdefs/MacroInstance.java (hunk 2 above)
Comment (bodewig) [reply src/main/org/apache/tools/ant/taskdefs/MacroInstance.java:225]: doesn't make me like it much more

File: src/main/org/apache/tools/ant/taskdefs/optional/jlink/ClassNameReader.java [hunk 6]
```diff
@@ -86,14 +87,9 @@
                 values[i] = data.readUnsignedShort();
                 break;
 
-            case FIELDREF :
-            case METHODREF :
-            case INTERFACEMETHODREF :
-            case NAMEANDTYPE :
-                values[i] = data.readInt();
-                break;
+            case UNUSED :
```
Comment (bodewig) [inline src/main/org/apache/tools/ant/taskdefs/optional/jlink/ClassNameReader.java:90]: let me note a pattern here, Personally I do not like cases that fall through to the `default` case.

File: src/main/org/apache/tools/ant/taskdefs/optional/jsp/WLJspc.java [hunk 7]
```diff
@@ -178,7 +178,7 @@ public void execute() throws BuildException {
                 args[j + 1] = destinationPackage;
             } else {
                 parents = this.replaceString(parents, File.separator, "_.");
-                args[j + 1] = destinationPackage + "." + "_" + parents;
+                args[j + 1] = destinationPackage + "._" + parents;
```
Comment (bodewig) [inline src/main/org/apache/tools/ant/taskdefs/optional/jsp/WLJspc.java:181]: please avoid mixing unrelated commits with your PRs.

File: src/main/org/apache/tools/ant/taskdefs/optional/net/FTP.java [hunk 8]
```diff
@@ -2690,6 +2680,8 @@ public int getAction() {
                 return RM_DIR;
             case "site":
                 return SITE_CMD;
+            case "send":
+            case "put":
```
Comment (bodewig) [inline src/main/org/apache/tools/ant/taskdefs/optional/net/FTP.java:2684]: see above ;-)
